addons/hr_timesheet/controllers/portal.py

```diff
@@ -117,7 +119,8 @@ def portal_my_timesheets(self, page=1, sortby=None, filterby=None, search=None,
             filterby = 'all'
         domain = AND([domain, searchbar_filters[filterby]['domain']])
 
-        if search and search_in:
+        if search:
+            search_in = search_in or 'all' # search_in could be '' and we want the 'all' filter in this case
```

Comment:
In which case would the `search_in` be just an empty string?

Reply:
It would be if you use a filter that isn't available in the filter dropdown ; see [here](https://github.com/odoo/odoo/blob/master/addons/portal/static/src/js/portal.js#L149)